rollup/sync_service.go

```diff
@@ -604,6 +586,25 @@ func (s *SyncService) maybeApplyTransaction(tx *types.Transaction) error {
 // Lower level API used to apply a transaction, must only be used with
 // transactions that came from L1.
 func (s *SyncService) applyTransaction(tx *types.Transaction) error {
+	meta := tx.GetMeta()
+	if tx.QueueOrigin().Uint64() == uint64(types.QueueOriginL1ToL2) {
+		s.SetLatestEnqueueIndex(meta.QueueIndex)
```

Comment:
Is there ever a situation in which `meta.QueueIndex` could be greater than the previous enqueue index?

Reply:
Do you mean less than? It should not
See the note below the `Types` section here: https://github.com/ethereum-optimism/specs/blob/main/transaction-ingestor.md#types